src/portable/portable.c

```diff
@@ -1911,32 +1941,31 @@ int portable_detach(
 
         /* Now, also drop any image symlink, for images outside of the sarch path */
         SET_FOREACH(item, markers) {
-                _cleanup_free_ char *sl = NULL;
+                _cleanup_free_ char *target = NULL;
                 struct stat st;
 
-                r = image_symlink(item, flags, &sl);
+                r = image_storage_path(item, flags, &target);
                 if (r < 0) {
-                        log_debug_errno(r, "Failed to determine image symlink for '%s', ignoring: %m", item);
+                        log_debug_errno(r, "Failed to determine image path for '%s', ignoring: %m", item);
                         continue;
                 }
 
-                if (lstat(sl, &st) < 0) {
-                        log_debug_errno(errno, "Failed to stat '%s', ignoring: %m", sl);
+                if (lstat(target, &st) < 0) {
+                        log_debug_errno(errno, "Failed to stat '%s', ignoring: %m", target);
                         continue;
                 }
 
-                if (!S_ISLNK(st.st_mode)) {
-                        log_debug("Image '%s' is not a symlink, ignoring.", sl);
-                        continue;
-                }
-
-                if (unlink(sl) < 0) {
-                        log_debug_errno(errno, "Can't remove image symlink '%s': %m", sl);
+                if (S_ISDIR(st.st_mode))
+                        r = rm_rf(target, REMOVE_ROOT | REMOVE_PHYSICAL | REMOVE_MISSING_OK | REMOVE_SYNCFS);
+                else
+                        r = RET_NERRNO(unlink(target));
```

Comment:
you can just use rm_rf() here always? why fall back to unlink()?

Reply:
it also allows you to drop the stat() and that will make the github code checker stop complaing btw.